docs/how-to-guides/create-a-user-defined-function.md

```diff
@@ -0,0 +1,913 @@
+    arrays correspond to the Java `List` type, not native Java arrays.
+
+Either continue following this guide by implementing more functions or skip ahead to [compiling the classes](#add-the-uberjar-to-the-classpath) so you can use the functions in ksqlDB.
+
+### Aggregation functions
+
+An aggregation function (UDAF for short) consumes one row at a time and maintains a stateful representation of all historical data. Use this when you want to compound data from multiple rows together.
+
+Create a file at `src/main/java/com/example/RollingSumUdaf.java` and populate it with the following code. This UDAF maintains a rolling sum of the last `3` integers in a stream, discarding the oldest values as new ones arrive.
+
+```java
+package my.example;
+
+import io.confluent.ksql.function.udaf.Udaf;
+import io.confluent.ksql.function.udaf.UdafDescription;
+import io.confluent.ksql.function.udaf.UdafFactory;
+import io.confluent.ksql.function.udf.UdfParameter;
+
+import java.util.List;
+import java.util.LinkedList;
+import java.util.Iterator;
+
+@UdafDescription(name = "rolling_sum",
+                 author = "example user",
+                 version = "2.0.0",
+                 description = "Maintains a rolling sum of the last 3 integers of a stream.")
+public class RollingSumUdaf {
+
+    private RollingSumUdaf() {
+    }
+
+    @UdafFactory(description = "Sums the previous 3 integers of a stream, discarding the oldest elements as new ones arrive.")
+    public static Udaf<Integer, List<Integer>, Integer> createUdaf() {
+        return new RollingSumUdafImpl();
+    }
+
+    private static class RollingSumUdafImpl implements Udaf<Integer, List<Integer>, Integer> {
+
+        private final int CAPACITY = 3;
+
+        @Override
+        public List<Integer> initialize() {
+            return new LinkedList<Integer>();
+        }
+
+        @Override
+        public List<Integer> aggregate(Integer newValue, List<Integer> aggregateValue) {
+            aggregateValue.add(newValue);
+
+            if (aggregateValue.size() > CAPACITY) {
+                aggregateValue = aggregateValue.subList(1, CAPACITY + 1);
+            }
+
+            return aggregateValue;
+        }
+
+        @Override
+        public Integer map(List<Integer> intermediate) {
+            Iterator<Integer> it = intermediate.iterator();
+            int k = 0;
+
+            while(it.hasNext()) {
+                k += it.next();
+            }
+
+            return k;
+        }
+
+        @Override
+        public List<Integer> merge(List<Integer> aggOne, List<Integer> aggTwo) {
+            return aggOne;
+        }
+    }
+}
+```
+
+There are many things to observe in this class:
+
+- By contrast to scalar and tabular functions, aggregation functions are designated by a static method with the `@UdafFactory` annotation. Because aggregations must implement multiple methods, this helps ksqlDB differentiate aggregations when multiple type signatures are used.
```

Comment:
```suggestion
- Aggregation functions are designated by a static method with the `@UdafFactory` annotation, which differs from scalar and tabular functions. Because aggregations must implement multiple methods, this helps ksqlDB differentiate aggregations when multiple type signatures are used.
```